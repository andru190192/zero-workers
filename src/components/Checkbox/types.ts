export interface Country {
  id: number;
  name: string;
}

export interface CheckBoxProps {
  label: string;
  value: number;
  selectedItems?: any[];
  handleOnChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}