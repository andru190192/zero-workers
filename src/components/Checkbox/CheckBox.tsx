import React from 'react';
import { CheckBoxProps } from './types';

const CheckBox: React.FC<CheckBoxProps> = ({ label, value, selectedItems, handleOnChange }) => {

  return (
    <div>
      <input
        id={label}
        name={label}
        value={value}
        type="checkbox"
        checked={selectedItems?.some((item) => item.id === value)}
        onChange={handleOnChange}
      />
      <label htmlFor={label}>{label}</label>
    </div>
  );
};

export default CheckBox;