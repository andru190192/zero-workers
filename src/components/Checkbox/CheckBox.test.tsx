import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import CheckBox from './CheckBox';
import { CheckBoxProps } from './types';

const handleOnChangeMock = jest.fn();

const defaultProps: CheckBoxProps = {
  label: 'USA',
  value: 1,
  selectedItems: [],
  handleOnChange: handleOnChangeMock,
}

describe('<CheckBox />', () => {

  test('should renders component correctly', () => {
    const { container } = render(<CheckBox {...defaultProps} />);
    const label = screen.getByText(/USA/i);
    const checkbox = screen.getByRole('checkbox');

    expect(label).toBeInTheDocument();
    expect(checkbox).not.toBeChecked();
    expect(container).toMatchSnapshot();
  });

  test('should calls handleOnChange prop when checkbox is clicked', () => {
    render(<CheckBox {...defaultProps} />);
    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);

    expect(handleOnChangeMock).toHaveBeenCalledTimes(1);
  });

  test('should checkbox is checked when selectedItems contains value', () => {
    const selectedItems = [{ id: 1, name: 'USA' }];
    render(<CheckBox {...defaultProps} selectedItems={selectedItems} />);
    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).toBeChecked();
  });

  test('should checkbox is not checked when selectedItems does not contain value', () => {
    const selectedItems = [{ id: '2', name: 'Ecuador' }];
    render(<CheckBox {...defaultProps} selectedItems={selectedItems} />);
    const checkbox = screen.getByRole('checkbox');
    
    expect(checkbox).not.toBeChecked();
  });
});