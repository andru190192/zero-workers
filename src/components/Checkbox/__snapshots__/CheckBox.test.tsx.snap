// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should renders component correctly 1`] = `
<div>
  <div>
    <input
      id="USA"
      name="USA"
      type="checkbox"
      value="1"
    />
    <label
      for="USA"
    >
      USA
    </label>
  </div>
</div>
`;
